fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/utils/impl/CacheEntryFactory.java

```diff
@@ -47,15 +53,21 @@ private CacheEntryFactory() {
      * @throws RepositoryException if repository exception occurred
      */
     public static CacheEntry forProperty(final Property property) throws RepositoryException {
-        if (property.getValue().getString().indexOf((MessageExternalBodyContentType.MEDIA_TYPE)) >= 0) {
+        // if it's an external binary, catch that here and treat it differently.
+        LOGGER.info("Did you know that property.getName() is: '{}' and Proxy: {} ", property.getName(),
+                PROXY_FOR.getNameSpace() + ":" + PROXY_FOR.getLocalName());
+        if (property.getName().endsWith(PROXY_FOR.getLocalName()) ||
```

Comment:
Since this is checking the JCR property, it will have a name like `fedora:redirectsTo`, right? I think you can switch to:
`if (property.getName().equals(FedoraTypes.PROXY_FOR))` and the same with REDIRECTS_TO rather than using the RDFLexicon version of the same properties.